### PR TITLE
Support bem-forum@v1.0.0

### PR DESCRIPTION
**Pull request by tavriaforever**
@tadatuta 
@tormozz48

### CODE REVIEW

File: src/blocks/common.blocks/root/root.bemtree [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        var ctx = this.ctx,
+            context = ctx.context;
+
+        this.data = ctx.data;
+
+        if (context) {
+            return applyCtx(context);
+        }
+
+        return applyCtx({ block: 'page' });
```
Comment (gela-d) [inline src/blocks/common.blocks/root/root.bemtree]: `return applyCtx(context ? context : { block: 'page' })`

File: src/blocks/server.blocks/middleware/__page/middleware__page.node.js [hunk 2]
```diff
@@ -83,17 +83,21 @@ modules.define('middleware__page', ['config', 'logger', 'util', 'model', 'templa
...
                         ctx = {
-                            block: 'i-global',
-                            req: req, // request object // TODO remove it and fix templates
-                            bundleName: 'common',
-                            lang: req.lang, // selected language
-                            statics: ''
+                            block: 'root',
+                            data: {
+                                title: res.locals.pageTitle,
+                                forum: res.locals,
```
Comment (gela-d) [inline src/blocks/server.blocks/middleware/__page/middleware__page.node.js]: var locals = res.locals

File: src/blocks/common.blocks/root/root.deps.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+({
+    mustDeps: [
+        { block: 'i-bem' }
```
Comment (tadatuta) [inline src/blocks/common.blocks/root/root.deps.js]: looks inconsistent with `page` as a string

**Comment by tadatuta** [comment]
have no idea what's going on here but syntax is fine ;)

File: src/blocks/common.blocks/root/root.bemtree (hunk 1 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/root/root.bemtree]: agree

File: src/blocks/server.blocks/middleware/__page/middleware__page.node.js (hunk 2 above)
Comment (tavriaforever) [inline src/blocks/server.blocks/middleware/__page/middleware__page.node.js]: thanks!

File: src/blocks/common.blocks/root/root.deps.js (hunk 3 above)
Comment (tavriaforever) [inline src/blocks/common.blocks/root/root.deps.js]: agree

**Pull request merged**